### PR TITLE
fix: [CDS-68201]: UX enhancements for re-run/re-try pipeline

### PR DESCRIPTION
**Pull request by PratyushGar**
### Summary

JIRA : https://harness.atlassian.net/browse/CDS-68201
Change list:
- Retry failed option removed from pipeline execution page kebab menu
- Retry failed pipeline renamed to Re-run Specific stages
- Retry failed pipeline option combined with re-run pipeline under dropdown with 
    - From Last Failed Stage ( pre-select last stage to re-execute pipeline )
    - From Selective Stage ( let the user choose a stage to resume from ) 
    - Entire pipeline ( re-run the pipeline )
- Successful Execution
image

- Re-try pipeline allowed scenario
image

- Listing page kebab menu 
image

#### Screenshots

https://github.com/harness/harness-core-ui/assets/100121280/67e36971-572e-4137-bd69-241e69960197

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by rameshx — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx [hunk 1]
```diff
@@ -224,6 +241,44 @@ export function ExecutionHeader({ pipelineMetadata }: ExecutionHeaderProps): Rea
...
+      disabled={!canExecute || isPipelineInvalid}
+    >
+      <RbacButton
+        {...commonReRunProps}
+        rightIcon="caret-down"
+        onClick={noop}
+        featuresProps={getFeaturePropsForRunPipelineButton({ modules, getString })}
+      />
+    </Popover>
+  )
```
Comment (udayvunnam) [inline src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx:280]: The Button UI looks different than the split button, why can't we use SplitButton

Comment (udayvunnam) [reply src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx:280]: We can extend it if there is a missing functionality, but with this we have two slightly different UX for button with dropdown

Comment (PratyushGar) [reply src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx:280]: SplitButton is not needed here as we don't want to have a primary control for button and menu appearing from split button caret, just a dropdown with options similar to SplitButtonOption. we have similar use cases with template listing and input set options as well.

**Review by udayvunnam — APPROVED** [review]

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx (hunk 1 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx:280]: Can we discuss of having a primary control? We can potentially save a click. A SplitButton makes sense here.

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx (hunk 1 above)
Comment (PratyushGar) [reply src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx:280]: I have discussed with PM and UX, SplitButton will reduce the visibility of (retry option, now in line with re-run), so the product agreed dropdown. Details here https://harness.atlassian.net/browse/CDS-67703?focusedCommentId=495227

**Pull request merged**